httpmon.cc

```diff
@@ -266,20 +270,34 @@ int httpClientMain(int id, ClientControl &control, ClientData &data)
 			if (control.compressed) {
 				headers = curl_slist_append(headers, "Accept-Encoding: gzip");
 			}
+			if (!control.header.empty()) {
+			    headers = curl_slist_append(headers, control.header.c_str());
+			}
 			curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
 			curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, false);
 			curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0);
 			curl_easy_setopt(curl, CURLOPT_SSL_VERIFYSTATUS, false);
 
+			/* configure POST  if needed */
+            if (control.post || !control.body.empty()){
+                curl_easy_setopt(curl, CURLOPT_POSTFIELDS, control.body);
+            }
 			/* Send HTTP request */
 			{
 				std::lock_guard<std::mutex> lock(data.mutex);
 				data.queueLength++;
 			}
 			responseFlags = 0;
-			if (timeout > 0) {
-				requestData.error = (curl_easy_perform(curl) != 0);
+            CURLcode res;
+            if (timeout > 0) {
+                res = curl_easy_perform(curl);
```

Comment:
`curl_easy_perform()` seems to be called twice. Wouldn't that generate two HTTP requests?